src/auth/session/storage/__tests__/battery-of-tests.ts

```diff
@@ -20,13 +25,77 @@ export function batteryOfTests(storageFactory: () => Promise<SessionStorage>) {
     await expect(storage.deleteSession(sessionId)).resolves.toBe(true);
   });
 
+  it('can store and delete offline sessions with expiry date', async () => {
+    const storage = await storageFactory();
+    const sessionId = 'test_session';
+    const session = new Session(sessionId, 'shop', 'state', false);
+    const expiryDate = new Date();
+    expiryDate.setMinutes(expiryDate.getMinutes() + 60);
+    session.expires = expiryDate;
+
+    await expect(storage.storeSession(session)).resolves.toBe(true);
+    expect(sessionEqual(await storage.loadSession(sessionId), session)).toBe(
+      true,
+    );
+
+    await expect(storage.storeSession(session)).resolves.toBe(true);
+    expect(sessionEqual(await storage.loadSession(sessionId), session)).toBe(
+      true,
+    );
+
+    await expect(storage.deleteSession(sessionId)).resolves.toBe(true);
+    await expect(storage.loadSession(sessionId)).resolves.toBeUndefined();
+
+    // Deleting a non-existing session should work
+    await expect(storage.deleteSession(sessionId)).resolves.toBe(true);
+  });
+
+  it('can store and delete offline sessions with user info', async () => {
```

Comment:
Technically, offline sessions won't ever have user info, but I don't think we need any special code to block saving the sessions in that case.